hier_config/platforms/functions.py

```diff
@@ -13,3 +13,47 @@ def expand_range(number_range_str: str) -> tuple[int, ...]:
         message = "len(set(numbers)) must be equal to len(numbers)."
         raise ValueError(message)
     return tuple(numbers)
+
+
+def convert_to_set_commands(config_raw: str) -> str:
+    """Convert a Juniper style config string into a list of set commands.
+
+    Args:
+        config_raw (str): The config string to convert to set commands
+    Returns:
+        config_raw (str): Configuration string
+
+    """
+    lines = config_raw.split("\n")
+    path: list[str] = []
+    set_commands: list[str] = []
+
+    for line in lines:
+        stripped_line = line.strip()
+
+        # Skip empty lines
+        if not stripped_line:
+            continue
+
+        # Strip ; from the end of the line
+        if stripped_line.endswith(";"):
+            stripped_line = stripped_line.replace(";", "")
+
+        # Count the number of spaces at the beginning to determine the level
+        level = line.find(stripped_line) // 4
+
+        # Adjust the current path based on the level
+        path = path[:level]
+
+        # If the line ends with '{' or '}', it starts a new block
+        if stripped_line.endswith(("{", "}")):
+            path.append(stripped_line[:-1].strip())
+        elif stripped_line.startswith(("set", "delete")):
+            # It's already a set command, so just add it to the list
+            set_commands.append(stripped_line)
+        else:
+            # It's a command line, construct the full command
+            command = f"set {' '.join(path)} {stripped_line}"
```

Comment:
Should this function take in the Driver's `declaration_prefix` instead of statically assuming it?

Reply:
It's assumed that we're injesting a JunOS style config:

```
system {
    host-name aggr-example.rtr;
}
```

And converting it to the set syntax:

```
set system host-name aggr-example.rtr
```

The JUNIPER_JUNOS and the VYOS drivers are currently the only operating systems that are using it.

Reply:
This is a nitpick and and I see both sides. So, I'm good with either approach. Passing in `declaration_prefix` eliminates the very minor repetition of "set " and keeps the behavior aligned with the driver for future use cases but adds a small amount of complexity.